### PR TITLE
Add graceful shutdown

### PR DESCRIPTION
**Pull request by juanarbol**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -786,12 +786,26 @@ if (opts.version) {
...
+    process.exit(code);
+  };
 }
 
 main()
-    .then(() => null)
     .catch(err => {
         logger.error('Top-level error (shutting down):', err);
-        process.exit(1);
+        terminationHandler('Top-level error', 0)(err);
```
Comment (apmorton) [inline app.js:810]: As noted, would prefer this remain `process.exit`, but if this line does stay the exit code must remain 1

Comment (juanarbol) [reply app.js:810]: I can remove that, to be like this:

```suggestion
        process.exit(1);
```

**Comment by partouf** [comment]
Perhaps a bit of context is in order here, what problem does this exactly solve?

**Comment by juanarbol** [comment]
> Perhaps a bit of context is in order here, what problem does this exactly solve?

Not a problem solver. This is considered good practice in Node.js apps

File: app.js (hunk 1 above)
Comment (juanarbol) [reply app.js]: Fixed! Thanks for reviewing!

**Review by mattgodbolt — APPROVED** [review]
Thanks!

**Comment by mattgodbolt** [comment]
This is now live

**Pull request merged**